File: gachisa-backend/src/main/java/com/gachisa/category/repository/CategoryRepository.java
package com.gachisa.category.repository;

import com.gachisa.category.entity.Category;
import java.util.List;
import org.springframework.data.jpa.repository.JpaRepository;

public interface CategoryRepository extends JpaRepository<Category, Long> {

    List<Category> findByParentIsNull();

    boolean existsByName(String name);
}

File: gachisa-backend/src/main/kotlin/com/gachisa/category/repository/CategoryRepository.kt
package com.gachisa.category.repository

import com.gachisa.category.entity.Category
import org.springframework.data.jpa.repository.JpaRepository

interface CategoryRepository : JpaRepository<Category, Long> {

    fun findByParentIsNull(): List<Category>

    fun existsByName(name: String): Boolean
}
